src/bytes_mut.rs

```diff
@@ -613,7 +613,7 @@ impl BytesMut {
         }
 
         debug_assert_eq!(kind, KIND_ARC);
-        let shared: *mut Shared = self.data as _;
+        let shared: *mut Shared = self.data;
```

Comment:
nit, take it or leave it: since there isn't a cast here any longer, can't the type annotation be removed?
```suggestion
        let shared = self.data;
```